numba/cuda/tests/cudapy/test_ufuncs.py

```diff
@@ -37,16 +37,31 @@ def setUp(self):
             (np.complex64(0.0), types.complex64),
             (np.complex64(0.5 + 0.5j), types.complex64),
 
-            (np.complex64(-0.5 - 0.5j), types.complex128),
-            (np.complex64(0.0), types.complex128),
-            (np.complex64(0.5 + 0.5j), types.complex128),
+            (np.complex128(-0.5 - 0.5j), types.complex128),
+            (np.complex128(0.0), types.complex128),
+            (np.complex128(0.5 + 0.5j), types.complex128),
 
             (np.array([-0.5 - 0.5j, 0.0, 0.5 + 0.5j], dtype='c8'),
              types.Array(types.complex64, 1, 'C')),
             (np.array([-0.5 - 0.5j, 0.0, 0.5 + 0.5j], dtype='c16'),
              types.Array(types.complex128, 1, 'C')),
         ])
 
+        # Test with multiple dimensions
+        self.inputs.extend([
+            # Basic 2D and 3D arrays
+            (np.linspace(0, 1).reshape((5, -1)),
+             types.Array(types.float64, 2, 'C')),
+            (np.linspace(0, 1).reshape((2, 5, -1)),
+             types.Array(types.float64, 3, 'C')),
+            # Complex data (i.e. interleaved)
+            (np.linspace(0, 1 + 1j).reshape(5, -1),
+             types.Array(types.complex128, 2, 'C')),
+            # F-ordered
+            (np.asfortranarray(np.linspace(0, 1).reshape((5, -1))),
+             types.Array(types.float64, 2, 'F')),
+        ])
```

Comment:
I couldn't get one in because the CUDA target won't copy a strided array to the device - instead I have tried to cover differently-strided cases by using a complex, and an F-ordered array. This would be possible with bigger-scale changes to the test harness, but I was trying to avoid changing too much of it.